### ISSUE TITLE
ansi rendering fails unless a 1ms delay is inserted into echo statement

### ISSUE THREAD

**Issue by ClaireCJS** [microsoft/terminal#18091]
### Windows Terminal version

1.20.11781.0

### Windows build number

10.0.19045.5011

### Steps to reproduce

I think this may be a new bug that just creep in. 

BACKGROUND: I have an "askyn.bat" that asks Yes/no questions. It has a "big" option to use VT100 double-height text.  Starting this week, my "big" double-height questions started getting a newline inserted in between.... But only if I was at the very lowest row of the console when calling it. It looked like this:
![Image](https://github.com/user-attachments/assets/e7e91b19-4f22-400b-9f4b-47b6a072ca64)

THE FIX: After some rather extensive testing, the solution was the very last thing I'd ever expect.

**It was inserting a 1ms delay prior to the newline!!!!!!!!**

![Image](https://github.com/user-attachments/assets/ae378d58-eb87-4683-b4eb-39aa44426e11)

Literally, remarking out the ```delay /m 1``` (one millisecond delay) changes the rendering from aforementioned screenshot to correct output:
![Image](https://github.com/user-attachments/assets/170bb39c-5450-4e60-98f8-6aff6b2b9f17)

I sincerely doubt this is by design.

I sincerely doubt that a 1 millisecond delay before outputting a newline should somehow change the ANSI rendering such that there is a phantom extra newline that appears out of nowhere.

And one of the reasons for my sincere doubt is —— This is a script i've been using for a year. It just broke today. Consistently.

FOR REFERENCE: "echos" is the same as echo, but without a newline at the end. "delay" is like unix's sleep, but accepts millisecond delays with the /m option. "big_top" and "big_bot" are the ansi codes for double-height text, top line and bottom line.  Ansi_clear_to_end is simply to clear any screen garbage off from the right of the line. Ansi_save_position is the ansi code to save screen position, which I use to write "yes" "no" in the correct place once the user has responded to the Y/N prompt.

### Expected Behavior

(See above)

### Actual Behavior

(see above)

**Comment by ClaireCJS** [comment]
This was a very interesting thing to discover today!

In 40 years of working with console stuff, I don't think I've ever had to interrupt text output with a 1 millisecond delay to get it to render correctly.

**Comment by lhecker** [comment]
That's indeed quite curious... Could you provide us with a minimal .bat file that causes the issue?

**Comment by ClaireCJS** [comment]
> That's indeed quite curious... Could you provide us with a minimal .bat file that causes the issue?

If you could tell me how to echo an ESC character from cmd.exe at the command line?  Otherwise, it's gonna technically be a .btm file (still named .bat though) intended for the TCC command line (jpsoft.com), in which i use %@char[27]

**Comment by DHowett** [comment]
You can insert a literal ESC in a batch file. :)

**Comment by ClaireCJS** [comment]
This bug may be too obscure anyway. More obscure than I initially reported. It's a more complicated scenario than I originally described to get it to trigger. It also involves moving up a couple of lines prior.  Let me try to replicate this in ```cmd.exe```

**Comment by ClaireCJS** [comment]
Is there a way in ```cmd.exe``` to echo without a newline at the end?

**Comment by ClaireCJS** [comment]
Okay!  Got one that works in cmd.exe!

To simulate the 1ms delay in my original test case scenario, I instead ```ping localhost >nul```

Anyway, here is ```bug-wc.bat```.  It displays the same text twice: 

1. without a delay, which   produces the bug
2. with  the delay, which suppresses the bug

```
@echo off
dir
dir
echo .
echo .
echo [1A[1A#3THE QUICK BROWN DOG JUMPED
rem    ping 127.0.0.1 -n 1 -w 1 >nul
echo #4THE QUICK BROWN DOG JUMPED 7[s

pause

dir
dir
echo .
echo .
echo [1A[1A#3THE QUICK BROWN DOG JUMPED
       ping 127.0.0.1 -n 1 -w 1 >nul
echo #4THE QUICK BROWN DOG JUMPED 7[s

```

**Comment by DHowett** [comment]
Huh. I can't reproduce this on 1.20, 1.21, 1.22 or 1.23.

Let's try this: can you grab an ETL trace following this guide? https://github.com/microsoft/terminal/wiki/Troubleshooting-Tips#capturing-a-debug-etl-trace

FYI: it will capture all output to _all console sessions_ produced during the capture session. If you are not comfortable sharing such a trace on GitHub, my e-mail is on my profile.

**Comment by ClaireCJS** [comment]
> Huh. I can't reproduce this on 1.20, 1.21, 1.22 or 1.23.
> 
> Let's try this: can you grab an ETL trace following this guide? https://github.com/microsoft/terminal/wiki/Troubleshooting-Tips#capturing-a-debug-etl-trace
> 
> FYI: it will capture all output to _all console sessions_ produced during the capture session. If you are not comfortable sharing such a trace on GitHub, my e-mail is on my profile.

Plot Twist: Bug only manifests if window is maximized.   Was yours maximized? {sorry!}

**Comment by ClaireCJS** [comment]
Another plot twist: This bug only manifests on 2 of my 4 displays.  And they aren't an expected combo either (a DVI monitor, and a Samsung tv, both in 1080p)

**Comment by ClaireCJS** [comment]
Alas, I'm having problems with the performance monitor.

```
C:\Users\ClioC>wpr -start GeneralProfile -start C:\bat\Terminal.wprp!DefTerm.Verbose
                                                                                                                                                                                                                     The profiles are already running.                                                                                                                                                                                                                                                                                                                                                                                                 Profile Id: GeneralProfile.Verbose.Memory, DefTerm.Verbose.Memory                                                                                                                                                                                                                                                                                                                                                                 Error code: 0xc5583001                                                                                                                                                                                                                                                                                                                                                                                                        

C:\Users\ClioC>wpr -stop terminal-trace.etl
                                                                                                                                                                                                                     Cannot change thread mode after it is set.                                                                                                                                                                                                                                                                                                                                                                                        Profile Id: RunningProfile 
Error code: 0x80010106
```

Seems like... I can't make the ETL trace happen for whatever reason [lovely error codes there].

**Comment by lhecker** [comment]
What happens if you enable this setting?

![Image](https://github.com/user-attachments/assets/9c0cfe2a-2dc1-4e1d-bca2-b45c6f72155e)

**Comment by j4james** [comment]
I suspect this is a bug in the old conpty implementation. If it's what I think it is, it likely only occurs when your screen width is an odd number of columns. It should be working correctly with the new conpty passthrough implementation.

**Comment by ClaireCJS** [comment]
> What happens if you enable this setting?
> 
> ![Image](https://github.com/user-attachments/assets/9c0cfe2a-2dc1-4e1d-bca2-b45c6f72155e)
> 
> (It's called "Disable partial Swap Chain invalidation" in newer versions.)

Same behavior

**Comment by ClaireCJS** [comment]
> I suspect this is a bug in the old conpty implementation. If it's what I think it is, it likely only occurs when your screen width is an odd number of columns. It should be working correctly with the new conpty passthrough implementation.

You're dead right!!!!  I checked. It is only with odd columns.  Which explains why it only manifested on ½ of my 4 monitors! 🤣

**Comment by DHowett** [comment]
Alright, given that this is gonna be fixed with 1.22+ I'm gonna close it out. Thanks all for the discussion!

**Comment by ClaireCJS** [comment]
I'm just glad it was a real bug with a real fix! :)
